app/src/main/java/io/homeassistant/companion/android/settings/language/LanguagesProvider.kt

```diff
@@ -0,0 +1,57 @@
+package io.homeassistant.companion.android.settings.language
+
+import android.content.Context
+import android.content.res.Configuration
+import android.content.res.Resources
+import android.os.Build
+import android.os.LocaleList
+import android.util.DisplayMetrics
+import io.homeassistant.companion.android.R
+import io.sentry.core.util.StringUtils
+import java.util.Locale
+
+class LanguagesProvider {
+
+    fun getSupportedLanguages(context: Context): Map<String, String> {
+        val listAppLocales = mutableMapOf<String, String>()
+
+        val resources = context.resources
+        val configuration = resources.configuration
+        val listLocales = resources.assets.locales
```

Comment:
Have no ideas why AssetManager ignores regional locales there, like zh-rCN, zh-rTW, fy-rNL. Any suggestion?

Reply:
My bad, it doesn't ignore. Moreover it return region locales we don't support dirrectly, like fr-CA(because we support fr) or es-US and I see no way to exclude them with current solution.